python/cudf/cudf/core/index.py

```diff
@@ -2861,3 +2836,72 @@ def _extended_gcd(a: int, b: int) -> Tuple[int, int, int]:
         old_s, s = s, old_s - quotient * s
         old_t, t = t, old_t - quotient * t
     return old_r, old_s, old_t
+
+
+def _get_indexer_basic(index, positions, method, target_col, tolerance):
+    nonexact = positions.isnull()
+    positions[nonexact] = index.searchsorted(
+        target_col[nonexact],
+        side="left" if method in {"pad", "ffill"} else "right",
+    )
+    if method in {"pad", "ffill"}:
+        # searchsorted returns "indices into a sorted array such that,
+        # if the corresponding elements in v were inserted before the
+        # indices, the order of a would be preserved".
+        # Thus, we need to subtract 1 to find values to the left.
+        positions[nonexact] -= 1
+        # This also mapped not found values (values of 0 from
+        # np.searchsorted) to -1, which conveniently is also our
+        # sentinel for missing values
+    else:
+        # Mark indices to the right of the largest value as not found
+        positions[positions == len(index)] = -1
+
+    if tolerance is not None:
+        distance = abs(index[positions] - target_col)
+        return positions.where(distance <= tolerance, -1)
+    return positions
+
+
+def _get_nearest_indexer(index, positions, target_col, tolerance):
+    """
+    Get the indexer for the nearest index labels; requires an index with
+    values that can be subtracted from each other.
+    """
+    if not len(index):
+        return _get_indexer_basic(
+            index=index,
+            positions=positions.copy(deep=True),
+            method="pad",
+            targe_col=target_col,
+            tolerance=tolerance,
+        )
```

Comment:
What is going on here? If the `index` is empty, then surely the only possible return value is an array of length `target` with -1 everywhere?

Reply:
Yes, simplified this further in new code.